skimage/restoration/uft.py

```diff
@@ -390,7 +390,8 @@ def ir2tf(imp_resp, shape, dim=None, is_real=True):
     if not dim:
         dim = imp_resp.ndim
     # Zero padding and fill
-    irpadded = np.zeros(shape)
+    irpadded_dtype = imp_resp.dtype if imp_resp.dtype.kind == 'f' else float
```

Comment:
could you explain why `_float_types` is not applicable here?

Reply:
I think it is now and will update it. I had a different `_float_types` implementation initially that didn't match this behavior for integer types.

This is still subtly different in that for the code here float16 or float128 would be passed on unchanged, but that isn't really desirable or necessary since the FFT functions later on will end up casting to single or double precision anyways.